lib/ansible/playbook/task.py

```diff
@@ -286,8 +286,21 @@ def _parse_env_kv(k, v):
                     env[k] = templar.template(v, convert_bare=False)
                 except AnsibleUndefinedVariable as e:
                     error = to_native(e)
-                    if self.action in ('setup', 'gather_facts') and 'ansible_facts.env' in error or 'ansible_env' in error:
-                        # ignore as fact gathering is required for 'env' facts
+                    plugins_ignore_undefined = (
+                        'add_host',
+                        'debug',
+                        'gather_facts',
+                        'group_by',
+                        'include_vars',
+                        'set_fact',
+                        'setup',
```

Comment:
this list can get long .. meta? include_role/include_tasks? .. sometimes 'include' ... shoudl really only affect initial fact gathering

Reply:
If listing exclusions here is not the best way to solve this, could we just warn and suggest using `| default` instead?

Reply:
agreed, only setup/gather_facts are normally handles since they make up 'implied task you cannot easily modify'

Reply:
also, they are the ones that CREATE the ansible_env the use